Core.ApplicationServices/SystemUsage/Write/ItSystemUsageWriteService.cs

```diff
@@ -163,20 +172,135 @@ public class ItSystemUsageWriteService : IItSystemUsageWriteService
                     var result = _referenceService.AddReference(systemUsage.Id, ReferenceRootType.SystemUsage, referenceProperties.Title, referenceProperties.DocumentId, referenceProperties.Url);
 
                     if (result.Failed)
-                        return new OperationError($"Failed to add reference with data:{JsonConvert.SerializeObject(referenceProperties)}. Error:{result.Error.Message.GetValueOrFallback(string.Empty)}", result.Error.FailureType);
+                        return new OperationError($"Failed to add reference with data:{JsonConvert.SerializeObject(referenceProperties)}. Error:{result.Error.Message.GetValueOrEmptyString()}", result.Error.FailureType);
 
                     if (referenceProperties.MasterReference)
                     {
                         var masterReferenceResult = systemUsage.SetMasterReference(result.Value);
                         if (masterReferenceResult.Failed)
-                            return new OperationError($"Failed while setting the master reference:{masterReferenceResult.Error.Message.GetValueOrFallback(string.Empty)}", masterReferenceResult.Error.FailureType);
+                            return new OperationError($"Failed while setting the master reference:{masterReferenceResult.Error.Message.GetValueOrEmptyString()}", masterReferenceResult.Error.FailureType);
                     }
                 }
             }
 
             return systemUsage;
         }
 
+        private Result<ItSystemUsage, OperationError> PerformArchivingUpdate(ItSystemUsage itSystemUsage, UpdatedSystemUsageArchivingParameters archivingProperties)
+        {
+            return WithOptionalUpdate(itSystemUsage, archivingProperties.ArchiveDuty, (usage, archiveDuty) => usage.ArchiveDuty = archiveDuty)
+                .Bind(systemUsage => WithOptionalUpdate(systemUsage, archivingProperties.ArchiveTypeUuid, UpdateArchiveType))
+                .Bind(systemUsage => WithOptionalUpdate(systemUsage, archivingProperties.ArchiveLocationUuid, UpdateArchiveLocation))
+                .Bind(systemUsage => WithOptionalUpdate(systemUsage, archivingProperties.ArchiveTestLocationUuid, UpdateArchiveTestLocation))
+                .Bind(systemUsage => WithOptionalUpdate(systemUsage, archivingProperties.ArchiveSupplierOrganizationUuid, UpdateArchiveSupplierOrganization))
+                .Bind(systemUsage => WithOptionalUpdate(systemUsage, archivingProperties.ArchiveActive, (usage, archiveActive) => usage.ArchiveFromSystem = archiveActive.HasValue ? archiveActive.Value : null))
+                .Bind(systemUsage => WithOptionalUpdate(systemUsage, archivingProperties.ArchiveNotes, (usage, archiveNotes) => usage.ArchiveNotes = archiveNotes))
+                .Bind(systemUsage => WithOptionalUpdate(systemUsage, archivingProperties.ArchiveFrequencyInMonths, (usage, archiveFrequencyInMonths) => usage.ArchiveFreq = archiveFrequencyInMonths.HasValue ? archiveFrequencyInMonths.Value : null))
```

Comment:
Brug match().. den tillader at OnNone() afleverer null modsat getvalueorfallback som stædit holder fast i non-.null

Reply:
Ændrede properties til at være nullable ligesom de er i modellen. Så er der ikke behov for match